src/state/nftMarket/types.ts

```diff
@@ -22,6 +22,9 @@ export interface State {
   data: {
     collections: Record<string, Collection> // string is the address
     nfts: Record<string, NftToken[]> // string is the collection address
+    isFetchingMoreNfts: boolean
```

Comment:
several new top-level flags here
Probably can be organised better, feel free to suggest.